engine/gamesys/src/gamesys/resources/res_collision_object.cpp

```diff
@@ -275,6 +282,16 @@ namespace dmGameSystem
                     dmPhysics::DeleteCollisionShape2D(resource->m_Shapes2D[i]);
                 }
             }
+            if (physics_context->m_3D)
```

Comment:
No need to check. It's valid to free a null pointer.
```
free(resource->m_Shapes2D);
free(resource->m_Shapes3D);
```

But never mind it here, the whole physics system is this mess of if-statements, checking 2d vs 3d :/